tests/integration/test_charm.py

```diff
@@ -326,10 +307,48 @@ async def test_s3_conf(ops_test: OpsTest, app: Application, s3_url: str):
     response = s3_client.list_buckets()
     bucket_list = [*map(lambda a: a["Name"], response["Buckets"])]
 
-    assert s3_bucket in bucket_list
+    assert s3_conf["bucket"] in bucket_list
 
     # Check content has been uploaded in the bucket
-    response = s3_client.list_objects(Bucket=s3_bucket)
+    response = s3_client.list_objects(Bucket=s3_conf["bucket"])
     object_count = sum(1 for _ in response["Contents"])
 
     assert object_count > 0
+
+    # Cleanup
```

Comment:
Cleanup so that subsequent tests (like chaos tests) are executed properly not against a mock S3 (and they would fail).